### PR TITLE
Fix expanding

### PR DESCRIPTION
**Pull request by hAckdamDys**
fixed some bugs in expanding, calendar now should work, also added possibility to change deadline for given todo

### CODE REVIEW

File: client/components/List/index.js [hunk 1]
```diff
@@ -115,9 +115,12 @@ class List extends Component {
      };
 
     selectCategory = (categoryId, todo) => {
+      categoryId = parseInt(categoryId)
```
Comment (chudy1997) [inline client/components/List/index.js:127]: Why did you add that ? You never use that.

File: server/server.js [hunk 2]
```diff
@@ -100,13 +100,6 @@ function createServer() {
     });
   });
 
-  server.post('/todos/changeCategory', (req, res) => {
-    res.setHeader('Access-Control-Allow-Origin', '*');
-    const {todoId, categoryId} = req.query;
-
-    db.changeCategory(todoId, categoryId).then(() => res.status(200).send('Category deleted successfully'));
```
Comment (chudy1997) [inline server/server.js:107]: We should return 200, when we successfully update category. Change it in '/changeCategory' which you left.

File: client/components/List/index.css [hunk 3]
```diff
@@ -82,13 +82,13 @@
 .list-router .buttonstyle {
     margin: 0px 10px 10px 10px;
     float: right;
-    border-radius: 70%;
-    background-color:#599bb3;
+    /*border-radius: 70%;*/
```
Comment (chudy1997) [inline client/components/List/index.css:85]: Remove this comment.

File: client/components/List/index.css [hunk 4]
```diff
@@ -109,4 +109,14 @@
     border: none;
     margin: 0 10px;
     border-bottom: 3px dashed #ADADAD;
-}
\ No newline at end of file
+}
+
+.list-router select{
+    /*margin: 3px 3px 3px 4px;*/
```
Comment (chudy1997) [inline client/components/List/index.css:115]: Remove comment.

File: server/dbConf.json [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {
     "host": "127.0.0.1",
-    "user": "root",
-    "password": "pass",
-    "database": "todos",
+    "user": "karol",
```
Comment (chudy1997) [inline server/dbConf.json:3]: Why have you changed this file ?

**Review by chudy1997 — CHANGES_REQUESTED** [review]

File: client/components/List/index.js (hunk 1 above)
Comment (hAckdamDys) [reply client/components/List/index.js:127]: categoryId = parseInt(categoryId)

      const todos = this.props.todos;
      const oldCategoryId = todo.categoryId;
      todo.categoryId = categoryId;

Comment (hAckdamDys) [reply client/components/List/index.js:127]: this was the bug I fixed, if you won't use parseInt then it is a string, and later it caused problems as categoryId in todos isn't string...
